### PR TITLE
Add Profile item types for StudyParticipant values

### PR DESCRIPTION
**Pull request by Erin-Mounts**
and for a table item type to segue to a profile subview (e.g. Settings, Health).

Also refactor tests to work with Bridge SDK test harness.

### CODE REVIEW

File: BridgeApp/BridgeApp/Profile/SBAProfileSection.swift [hunk 1]
```diff
@@ -548,3 +551,46 @@ public struct SBAResourceProfileTableItem: SBAProfileTableItem, Decodable, RSDRe
 
 }
  */
+
+/// A profile table item that, when selected, segues to another profile table view.
+public struct SBAProfileViewProfileTableItem: SBAProfileTableItem, Decodable {
+    private enum CodingKeys: String, CodingKey {
+        case title, detail, inCohorts, notInCohorts, _icon = "icon", profileDataSource
+    }
+    // MARK: SBAProfileTableItem
```
Comment (syoung-smallwisdom) [inline BridgeApp/BridgeApp/Profile/SBAProfileSection.swift]: Nit: missing spaces

File: BridgeApp/BridgeApp/Profile/SBAProfileSection.swift [hunk 2]
```diff
@@ -548,3 +551,46 @@ public struct SBAResourceProfileTableItem: SBAProfileTableItem, Decodable, RSDRe
...
+    
+    /// A set of cohorts (data groups) the participant must not be in, in order to show this item in its containing profile section.
+    public var notInCohorts: Set<String>?
+    
+    /// Profile view profile table items segue to their associated profile table view when selected.
+    public var onSelected: SBAProfileOnSelectedAction? {
+        return .showProfileView
+    }
+    
+    // MARK: Profile View Profile Table Item
```
Comment (syoung-smallwisdom) [inline BridgeApp/BridgeApp/Profile/SBAProfileSection.swift]: Nit: missing space

File: BridgeApp/BridgeApp/Profile/SBAProfileSection.swift (hunk 1 above)
Comment (Erin-Mounts) [reply BridgeApp/BridgeApp/Profile/SBAProfileSection.swift]: oops

File: BridgeApp/BridgeApp/Profile/SBAProfileSection.swift (hunk 2 above)
Comment (Erin-Mounts) [reply BridgeApp/BridgeApp/Profile/SBAProfileSection.swift]: thanks

**Pull request merged**